src/sirius_interface.F

```diff
@@ -420,54 +425,80 @@ SUBROUTINE cp_sirius_create_env(pwdft_env)
-                  CALL sirius_set_atom_type_hubbard(sctx, label, lu, nu, &
-                                                    occ_u, U_u, J_u, alpha_u, beta_u, J0_u)
-               ELSE
-                  CALL sirius_set_atom_type_hubbard(sctx, label, lu, nu, &
-                                                    occ_u, u_minus_j, 0.0_dp, alpha_u, beta_u, J0_u)
-               END IF
-            END IF
             IF (ASSOCIATED(density)) DEALLOCATE (density)
             IF (ASSOCIATED(wavefunction)) DEALLOCATE (wavefunction)
             IF (ASSOCIATED(wfninfo)) DEALLOCATE (wfninfo)
 
             CALL deallocate_grid_atom(atom_grid)
             DEALLOCATE (rp, fun)
             DEALLOCATE (gth_atompot)
-!
          ELSE
             CALL cp_abort(__LOCATION__, &
                           "CP2K/SIRIUS: atomic kind needs UPF or GTH potential definition")
          END IF
+
+         CALL get_qs_kind(qs_kind_set(ikind), &
+                          dft_plus_u_atom=dft_plus_u_atom, &
+                          l_of_dft_plus_u=lu, &
+                          n_of_dft_plus_u=nu, &
+                          u_minus_j_target=u_minus_j, &
+                          U_of_dft_plus_u=U_u, &
+                          J_of_dft_plus_u=J_u, &
+                          alpha_of_dft_plus_u=alpha_u, &
+                          beta_of_dft_plus_u=beta_u, &
+                          J0_of_dft_plus_u=J0_u, &
+                          occupation_of_dft_plus_u=occ_u)
+
+         IF (dft_plus_u_atom) THEN
+            IF (nu < 1) THEN
+               CPABORT("CP2K/SIRIUS (hubbard): principal quantum number not specified")
+            END IF
+
+            IF (lu < 0) THEN
+               CPABORT("CP2K/SIRIUS (hubbard): l can not be negative.")
+            END IF
+
+            IF (lu >= nu) THEN
+               CPABORT("CP2K/SIRIUS (hubbard): the angular momentum should be lower than the principal quantum number")
+            END IF
+
+            IF (occ_u < 0.0) THEN
+               CPABORT("CP2K/SIRIUS (hubbard): the occupation number can not be negative.")
+            END IF
+
+            IF (ABS(u_minus_j) < 1e-8) THEN
+               CALL sirius_set_atom_type_hubbard(sctx, label, lu, nu, &
+                                                 occ_u, U_u, J_u, alpha_u, beta_u, J0_u)
+            ELSE
+               CALL sirius_set_atom_type_hubbard(sctx, label, lu, nu, &
+                                                 occ_u, u_minus_j, 0.0_dp, alpha_u, beta_u, J0_u)
+            END IF
+         END IF
+         !
       END DO
 
+      IF (ASSOCIATED(dft_plus_u_section)) THEN
+         CALL section_vals_val_get(dft_plus_u_section, keyword_name="_SECTION_PARAMETERS_", l_val=use_u_plus_v)
+         CALL sirius_set_parameters(sctx, hubbard_correction=use_u_plus_v)
+         CALL section_vals_val_get(dft_plus_u_section, keyword_name="SUBSPACE_METHOD", i_val=subspace_method)
+         IF (subspace_method .EQ. 0) THEN
```

Comment:
Instead of hard-coding 0,1,2,3,4, maybe define constants (`PARAMETER`) instead.

Reply:
Well it means having the sirius module included in the most basic cp2k_input file. I can do this but then there will be ifdef statements.

Reply:
should be solved with the most recent commit